spec/access/access_control/AccessControl.behavior.ts

```diff
@@ -0,0 +1,169 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { describeFilter } from '@solidstate/library';
+import { AccessControl } from '@solidstate/typechain-types';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+
+const DEFAULT_ADMIN_ROLE =
+  '0x0000000000000000000000000000000000000000000000000000000000000000';
+const ROLE = ethers.utils.solidityKeccak256(['string'], ['ROLE']);
+
+interface AccessControlBehaviorArgs {
+  deploy: () => Promise<AccessControl>;
+  getAdmin: () => Promise<SignerWithAddress>;
+  getNonAdmin: () => Promise<SignerWithAddress>;
+}
+
+export function describeBehaviorOfAccessControl(
+  { deploy, getAdmin, getNonAdmin }: AccessControlBehaviorArgs,
+  skips?: string[],
+) {
+  const describe = describeFilter(skips);
+
+  describe('::AccessControl', function () {
+    let instance: AccessControl;
+    let admin: SignerWithAddress;
+    let nonAdmin: SignerWithAddress;
+
+    beforeEach(async function () {
+      instance = await deploy();
+      admin = await getAdmin();
+      nonAdmin = await getNonAdmin();
+    });
+
+    describe('#hasRole(bytes32, address)', function () {
+      it('returns whether given account has given role', async function () {
+        expect(
+          await instance.callStatic['hasRole(bytes32,address)'](
+            DEFAULT_ADMIN_ROLE,
+            admin.address,
+          ),
+        ).to.equal(true);
+
+        expect(
+          await instance.callStatic['hasRole(bytes32,address)'](
+            DEFAULT_ADMIN_ROLE,
+            ethers.constants.AddressZero,
+          ),
+        ).to.equal(false);
+      });
+    });
+
+    describe('#getRoleAdmin(bytes32)', function () {
+      it('returns default admin role', async function () {
+        expect(
+          await instance.callStatic['getRoleAdmin(bytes32)'](ROLE),
+        ).to.equal(DEFAULT_ADMIN_ROLE);
+      });
+
+      it('returns default admin role as admin of itself', async function () {
+        expect(
+          await instance.callStatic['getRoleAdmin(bytes32)'](
+            DEFAULT_ADMIN_ROLE,
+          ),
+        ).to.equal(DEFAULT_ADMIN_ROLE);
+      });
+    });
+
+    describe('#grantRole(bytes32,address)', function () {
+      it('adds role to account', async function () {
+        await instance.connect(admin).grantRole(ROLE, nonAdmin.address);
```

Comment:
The instance should already be connected to the admin signer. I noticed this in both the test files.

```suggestion
        await instance.grantRole(ROLE, nonAdmin.address);
```

Reply:
Reliance on default accounts can cause unexpected behavior when spec is run in an arbitrary end-user environment.  SS is moving towards specifying sender explicitly, but there may still be some inconsistencies.

Reply:
Good point, ive definitely lost track of which signer is attached to which instance before so being explicit is beneficial for everyone involved.